File: android/app/src/main/java/net/result/taulight/MemberClient.java
package net.result.taulight;

import android.annotation.TargetApi;
import android.os.Build;

import net.result.sandnode.serverclient.SandnodeClient;

import java.util.Optional;

public final class MemberClient {
    public String nickname;
    public final SandnodeClient client;
    public final String link;

    public MemberClient(SandnodeClient client, String link) {
        this.client = client;
        this.link = link;
    }

    @TargetApi(Build.VERSION_CODES.N)
    public Optional<String> nickname() {
        return Optional.ofNullable(nickname);
    }
}

File: android/app/src/main/java/net/result/taulight/MemberClient.kt
package net.result.taulight

import net.result.sandnode.link.SandnodeLinkRecord
import net.result.sandnode.serverclient.SandnodeClient

class MemberClient(val client: SandnodeClient, val link: SandnodeLinkRecord) {
    var nickname: String? = null
}
